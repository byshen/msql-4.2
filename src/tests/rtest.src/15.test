#
# Test the datetime type
#

create table datetime(v1 datetime)\p\g
create index idx1 on datetime(v1)\p\g

INSERT INTO datetime  VALUES ('26-Jun-2004 15:16:19')\p\g
INSERT INTO datetime  VALUES ('02-Dec-1999 23:10:00')\p\g
INSERT INTO datetime  VALUES ('06-Feb-2004 01:01:01')\p\g
INSERT INTO datetime  VALUES ('02-Dec-1999 23:09:50')\p\g
INSERT INTO datetime  VALUES ('25-Jun-2004 11:22:33')\p\g

select * from datetime order by v1\p\g
select * from datetime where v1 = '06-Feb-2004 01:01:01'\p\g
select * from datetime where v1 between '01-Jan-1999 00:00:00'
	and '31-Dec-1999 23:59:59'\p\g
