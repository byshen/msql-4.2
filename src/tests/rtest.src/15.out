
Welcome to the miniSQL monitor.  Type \h for help.


mSQL >     -> 
Query buffer
------------

create table datetime(v1 datetime)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
create index idx1 on datetime(v1)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
INSERT INTO datetime  VALUES ('26-Jun-2004 15:16:19')
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
INSERT INTO datetime  VALUES ('02-Dec-1999 23:10:00')
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
INSERT INTO datetime  VALUES ('06-Feb-2004 01:01:01')
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
INSERT INTO datetime  VALUES ('02-Dec-1999 23:09:50')
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
INSERT INTO datetime  VALUES ('25-Jun-2004 11:22:33')
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
select * from datetime order by v1
[continue]
    -> 
Query OK.  5 row(s) modified or retrieved.

 +----------------------+
 | v1                   |
 +----------------------+
 | 02-Dec-1999 23:09:50 |
 | 02-Dec-1999 23:10:00 |
 | 06-Feb-2004 01:01:01 |
 | 25-Jun-2004 11:22:33 |
 | 26-Jun-2004 15:16:19 |
 +----------------------+



mSQL > 
Query buffer
------------
select * from datetime where v1 = '06-Feb-2004 01:01:01'
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.

 +----------------------+
 | v1                   |
 +----------------------+
 | 06-Feb-2004 01:01:01 |
 +----------------------+



mSQL >     -> 
Query buffer
------------
select * from datetime where v1 between '01-Jan-1999 00:00:00'
	and '31-Dec-1999 23:59:59'
[continue]
    -> 
Query OK.  2 row(s) modified or retrieved.

 +----------------------+
 | v1                   |
 +----------------------+
 | 02-Dec-1999 23:09:50 |
 | 02-Dec-1999 23:10:00 |
 +----------------------+



mSQL > 
Bye!

